modules/moderation.py

```diff
@@ -629,126 +545,91 @@ async def mod_error(self, ctx, error):
+            except KeyError: # This is a rare edge case, but if a pun is manually created the user may not have the flag yet. More a dev handler than not
+                logging.error(f'[Moderation] Expiry failed. Could not get strike_check from db.users resolving for pun {_id}, was it manually added?')
 
-            if pun['type'] == 'mute' and pun['expiry']: # A mute that has an expiry, for member in currently in guild
-                if int(time.time()) < pun['expiry']: continue # Has not expired yet
+            # Start logic
+            if doc['active_strike_count'] - 1 == 0:
+                db.update_one({'_id': doc['_id']}, {'$set': {'active': False}, '$inc': {'active_strike_count': -1}})
+                strikes = [x for x in db.find({'user': doc['user'], 'type': 'strike', 'active': True}).sort('timestamp', 1)]
+                if not strikes: # Last active strike expired, no additional
+                    return
 
-                newPun = db.find_one_and_update({'_id': pun['_id']}, {'$set': {
-                    'active': False
-                }})
-                docID = await utils.issue_pun(member.id, self.bot.user.id, 'unmute', 'Mute expired', active=False, context=pun['_id'])
+                self.taskHandles.append(self.bot.loop.call_later(60 * 60 * 12, asyncio.create_task, self.expire_actions(strikes[0]['_id'], guild)))
 
-                if not newPun: # There is near zero reason this would ever hit, but in case...
-                    logging.error(f'[expiry_check] Database failed to update user on pun expiration of {pun["_id"]}')
-                    continue
+            elif doc['active_strike_count'] > 0:
+                db.update_one({'_id': doc['_id']}, {'$inc': {'active_strike_count': -1}})
+                self.taskHandles.append(self.bot.loop.call_later(60 * 60 * 12, asyncio.create_task, self.expire_actions(doc['_id'], guild)))
 
-                await member.remove_roles(self.roles[pun['type']])
-                try:
-                     await member.send(utils.format_pundm('unmute', 'Mute expired', None, auto=True))
+            else:
+                logging.warning(f'[Moderation] Expiry failed. Doc {_id} had a negative active strike count and was skipped')
+                return
 
-                except discord.Forbidden: # User has DMs off
-                    pass
+            userDB.update_one({'_id': doc['user']}, {'$set': {'strike_check': time.time() + 60 * 60 * 24 * 7}})
 
-                await utils.send_modlog(self.bot, self.modLogs, 'unmute', docID, 'Mute expired', user=member, moderator=self.bot.user, public=True)
+        elif doc['type'] == 'mute' and doc['expiry']: # A mute that has an expiry
+            if doc['active'] == False: return # Mute was set to inactive between checks
+            if doc['expiry'] > time.time(): # To prevent drift we recall every 12 hours. Schedule for 12hr or expiry time, whichever is sooner. 5 seconds is for drift lienency
```

Comment:
"5 seconds is for drift lienency" comment is unclear what it refers to

Reply:
Duplicate of https://github.com/rNintendoSwitch/MechaBowser/pull/37#discussion_r554703680